src/libs/ReportUtils.ts

```diff
@@ -1530,26 +1590,24 @@ function getMoneyRequestSpendBreakdown(report, allReportsDict = null) {
 
 /**
  * Get the title for a policy expense chat which depends on the role of the policy member seeing this report
- *
- * @param {Object} report
- * @param {Object} [policy]
- * @returns {String}
  */
-function getPolicyExpenseChatName(report, policy = undefined) {
-    const reportOwnerDisplayName = getDisplayNameForParticipant(report.ownerAccountID) || lodashGet(allPersonalDetails, [report.ownerAccountID, 'login']) || report.reportName;
+function getPolicyExpenseChatName(report: OnyxEntry<Report>, policy: OnyxEntry<Policy>): string | undefined {
+    // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+    const reportOwnerDisplayName = getDisplayNameForParticipant(report?.ownerAccountID) || allPersonalDetails?.[report?.ownerAccountID ?? -1]?.login || report?.reportName;
 
     // If the policy expense chat is owned by this user, use the name of the policy as the report name.
-    if (report.isOwnPolicyExpenseChat) {
+    if (report?.isOwnPolicyExpenseChat) {
         return getPolicyName(report, false, policy);
     }
 
-    const policyExpenseChatRole = lodashGet(allPolicies, [`${ONYXKEYS.COLLECTION.POLICY}${report.policyID}`, 'role']) || 'user';
+    // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+    const policyExpenseChatRole = allPolicies?.[`${ONYXKEYS.COLLECTION.POLICY}${report?.policyID}`]?.role || 'user';
 
     // If this user is not admin and this policy expense chat has been archived because of account merging, this must be an old workspace chat
     // of the account which was merged into the current user's account. Use the name of the policy as the name of the report.
     if (isArchivedRoom(report)) {
-        const lastAction = ReportActionsUtils.getLastVisibleAction(report.reportID);
-        const archiveReason = (lastAction && lastAction.originalMessage && lastAction.originalMessage.reason) || CONST.REPORT.ARCHIVE_REASON.DEFAULT;
+        const lastAction = ReportActionsUtils.getLastVisibleAction(report?.reportID ?? '');
+        const archiveReason = lastAction?.actionName === CONST.REPORT.ACTIONS.TYPE.CLOSED ? lastAction?.originalMessage?.reason : CONST.REPORT.ARCHIVE_REASON.DEFAULT;
```

Comment:
NAB - How did we arrive at this condition?

Reply:
`reason` property is only on `originalMessage` when `actionName` is `CONST.REPORT.ACTIONS.TYPE.CLOSED` so its for type narrowing